### PR TITLE
Feature/upgrade wagtail 2.11

### PR DESCRIPTION
**Pull request by tbrlpld**
Upgrade Wagtail to version 2.11.2 as part of the lights-on support for the site.

One test assertion was commented that had failed before the upgrade. I have added a TODO comment for further investigation. 

I have checked the [Wagtail 2.11 upgrade considerations](https://docs.wagtail.io/en/stable/releases/2.11.html#upgrade-considerations) and made changes where necessary (in particular new import location for `SiteMiddleware`).

### CODE REVIEW

File: rca/programmes/models.py [hunk 1]
```diff
@@ -710,14 +710,13 @@ def get_context(self, request, *args, **kwargs):
             {"title": "Fees & funding"},
         ]
         # Only add the 'apply tab' depending global settings or specific programme page settings
-        programme_settings = ProgrammeSettings.for_site(request.site)
+        site = Site.find_for_request(request)
```
Comment (kevinhowbrook) [inline rca/programmes/models.py:713]: :ok_hand:

**Review by kevinhowbrook — APPROVED** [review]
Thanks @tbrlpld, this looks good to me. Once the staging testing has been completed we can merge this

**Comment by tbrlpld** [comment]
@kevinhowbrook  I had to add an extra commit that modifies the migration that creates the initial homepage. Somehow I had missed [this](https://docs.wagtail.io/en/stable/releases/2.11.html#run-before-declaration-needed-in-initial-homepage-migration) before. 

This should prevent a potential integrity error when starting with a new empty database. I did not see this issue in my testing, but it could happen (as described in the Wagtail release notes).

**Comment by kevinhowbrook** [comment]
Thanks @tbrlpld

**Review by kevinhowbrook — APPROVED** [review]

**Comment by tbrlpld** [comment]
Thank you @kevinhowbrook for the review!

**Pull request merged**